packages/react-components/src/components/Picker/Picker.tsx

```diff
@@ -70,32 +77,72 @@ export const Picker: React.FC<IPickerProps> = ({
     }
   }, [isListOpen]);
 
-  const handleOnTriggerClick = () => {
-    if (disabled) {
+  const handleTrigger = (e: React.MouseEvent | KeyboardEvent) => {
+    const target = e.target as Element;
+
+    if (disabled || target.getAttribute('data-dismiss-icon')) {
       return;
     }
 
-    return setIsListOpen((prev) => !prev);
+    setIsListOpen((prev) => !prev);
   };
 
   const handleOnClose = () => {
     setIsListOpen(false);
   };
 
-  const handleOnSelect = (item: IPickerListItem) => {
+  const handleSelect = (item: IPickerListItem) => {
+    if (type === 'single') {
+      setIsListOpen(false);
+      return onSelect([item]);
+    }
+
+    const currentSelectedItems = selected;
+
+    if (!currentSelectedItems) {
+      return onSelect([item]);
+    }
+
+    const newSelectedItems = currentSelectedItems.includes(item)
+      ? currentSelectedItems.filter((selectedItem) => selectedItem !== item)
+      : currentSelectedItems.concat(item);
+
+    onSelect(newSelectedItems);
+  };
+
+  const isItemSelectable = (item: IPickerListItem) =>
+    !item.disabled && !item.groupHeader && item.key !== 'select-all';
+
+  const handleSelectAll = () => {
     setIsListOpen(false);
-    setSelectedItem(item);
-    onSelect(item);
+
+    const itemsToSelect = items.filter(isItemSelectable);
+
+    onSelect(itemsToSelect);
   };
 
-  const handleOnClearClick = () => {
+  const handleClear = () => {
     setIsListOpen(false);
-    setSelectedItem(null);
     onSelect(null);
   };
 
   const handleOnFilter = (text: string) => setSearchPhrase(text);
 
+  const handleItemRemove = (item: IPickerListItem) => {
+    const newSelectedItems = selected
+      ? selected.filter((selectedItem) => selectedItem !== item)
+      : null;
+
+    if (newSelectedItems?.length === 0) {
+      return onSelect(null);
+    }
+
+    onSelect(newSelectedItems);
+  };
+
+  const getSelectedItemsKeys = (items: IPickerListItem[]) =>
+    items ? items.map((item) => item.key) : null;
```

Comment:
Consider the following approach
```suggestion
  const selectedItemsKeys = useMemo(() => {
    if (!selected || !items) {
      return null;
    }
    
    return items.map((item) => item.key);
  }, [selected, items]);
```